### PR TITLE
VZ-2393: Align install-time user role bindings with project role bindings

### PR DESCRIPTION
**Pull request by wmhopkins**
# Description

Changes to align install-time cluster roles/bindings with how project role bindings are done:
- Use rbacv1.Subject to represent binding subjects.
- Align field names used in vz CR with those in project CR.
- Allow for multiple binding subjects to be specified in in the vz CR, rather than just one each for admin/monitor.

Also adds validation for subjects provided in the vz CR.

Fixes VZ-2393.

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [x] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: platform-operator/controllers/verrazzano/installjob/install_config.go [hunk 1]
```diff
@@ -572,46 +572,103 @@ func getVerrazzanoInstallArgs(vzSpec *installv1alpha1.VerrazzanoSpec) ([]Install
...
+					SetString: true,
+				})
+			}
+			if len(v.APIGroup) > 0 {
+				args = append(args, InstallArg{
+					Name:      fmt.Sprintf("security.monitorSubjects.subject-%d.apiGroup", i),
+					Value:     v.APIGroup,
+					SetString: true,
+				})
+			}
```
Comment (fredtibbitts) [inline platform-operator/controllers/verrazzano/installjob/install_config.go:668]: There seems to be a bunch of duplicate code here, can you refactor this into a function a remove the duplication?

File: platform-operator/controllers/verrazzano/installjob/install_config.go [hunk 2]
```diff
@@ -572,46 +572,103 @@ func getVerrazzanoInstallArgs(vzSpec *installv1alpha1.VerrazzanoSpec) ([]Install
...
-		if k == "" {
-			k = "Group"
-		}
-		if k != "Group" && k != "User" {
-			err := fmt.Errorf("Unsuppored subject kind %s", k)
-			return []InstallArg{}, err
+	if len(vzSpec.Security.AdminSubjects) > 0 {
+		for i, v := range vzSpec.Security.AdminSubjects {
+			if len(v.Name) < 1 {
+				err := fmt.Errorf("No name for adminSubjects[%d]", i)
```
Comment (fredtibbitts) [inline platform-operator/controllers/verrazzano/installjob/install_config.go:578]: Golang errors are not supposed to start with a capital letter. I'm surprised the linter didn't catch this.

**Review by fredtibbitts — COMMENTED** [review]
Generally looks fine to me, just a couple of minor comments.

File: platform-operator/controllers/verrazzano/installjob/install_config.go (hunk 1 above)
Comment (wmhopkins) [reply platform-operator/controllers/verrazzano/installjob/install_config.go:670]: Yup, good idea.###

File: platform-operator/controllers/verrazzano/installjob/install_config.go (hunk 2 above)
Comment (wmhopkins) [reply platform-operator/controllers/verrazzano/installjob/install_config.go:580]: Will fix.

Comment (wmhopkins) [reply platform-operator/controllers/verrazzano/installjob/install_config.go]: Fixed.

File: platform-operator/controllers/verrazzano/installjob/install_config.go (hunk 1 above)
Comment (wmhopkins) [reply platform-operator/controllers/verrazzano/installjob/install_config.go:637]: Added a separate function.

**Review by fredtibbitts — APPROVED** [review]

**Pull request merged**